python/tank/util/shotgun.py

```diff
@@ -1157,117 +1233,151 @@ def _create_published_file(tk, context, path, name, version_number, task, commen
         # schemes are unlikely!
         if len(res.scheme) > 1 or not res.scheme.isalpha():
             path_is_url = True
-        
-    code = ""
+
+    # naming and path logic is different depending on url
     if path_is_url:
-        code = os.path.basename(res.path)
-    else:
-        code = os.path.basename(path)
 
-    # if the context does not have an entity, link it up to the project
-    if context.entity is None:
-        linked_entity = context.project
-    else:
-        linked_entity = context.entity
-    
-    data = {}
-    
-    # we set the optional additional fields first so we don't allow overwriting the standard parameters
-    if sg_fields is None:
-        sg_fields = {}
-    data.update(sg_fields)
-    
-    # standard parameters
-    data.update({
-        "code": code,
-        "description": comment,
-        "name": name,
-        "project": context.project,
-        "entity": linked_entity,
-        "task": task,
-        "version_number": version_number,
-        })
+        # extract name from url:
+        #
+        # scheme://hostname.com/path/to/file.ext -> file.ext
+        # scheme://hostname.com -> hostname.com
+        if res.path:
+            # scheme://hostname.com/path/to/file.ext -> file.ext
+            data["code"] = res.path.split("/")[-1]
+        else:
+            # scheme://hostname.com -> hostname.com
+            data["code"] = res.netloc
 
-    # handle the path definition
-    if path_is_url:
+        # make sure that the url is escaped property, otherwise
+        # shotgun might not accept it.
+        #
         # for quoting logic, see bugfix here:
         # http://svn.python.org/view/python/trunk/Lib/urllib.py?r1=71780&r2=71779&pathrev=71780
         #
-        # note: by appling a safe pattern like this, we guarantee that already quoted paths
+        # note: by applying a safe pattern like this, we guarantee that already quoted paths
         #       are not touched, e.g. quote('foo bar') == quote('foo%20bar')
-        data["path"] = {"url": urllib.quote(path, safe="%/:=&?~#+!$,;'@()*[]")}
+        data["path"] = {
+            "url": urllib.quote(path, safe="%/:=&?~#+!$,;'@()*[]"),
```

Comment:
it's taken from http://svn.python.org/view/python/trunk/Lib/urllib.py?r1=71780&r2=71779&pathrev=71780 (see above). Only used here and feels very specific to this particular quoting workflow - normally we would only use the std quoting methods in python, but they don't handle already quotes strings :)